atomic_reactor/utils/imageutil.py

```diff
@@ -5,6 +5,91 @@
 This software may be modified and distributed under the terms
 of the BSD license. See the LICENSE file for details.
 """
+
+import functools
+from typing import Optional, Union
+
+from osbs.utils import ImageName
+
+from atomic_reactor import config
+from atomic_reactor import util
+
+
+def inspectable(image: Union[str, ImageName]) -> bool:
+    """Check if we should expect the image to be inspectable."""
+    im = str(image)
+    return not (util.base_image_is_scratch(im) or util.base_image_is_custom(im))
+
+
+class ImageUtil:
+    """Convenience class for working with images relevant to the build process.
+
+    Supports e.g. inspecting the base image and other parent images.
+    """
+
+    def __init__(self, dockerfile_images: util.DockerfileImages, conf: config.Configuration):
+        """Init an ImageUtil.
+
+        :param dockerfile_images: information about the image references in the Dockerfile
+        :param conf: atomic-reactor configuration
+        """
+        self._dockerfile_images = dockerfile_images
+        self._conf = conf
+
+    def set_dockerfile_images(self, dockerfile_images: util.DockerfileImages) -> None:
+        """Set a new dockerfile_images instance."""
+        self._dockerfile_images = dockerfile_images
+
+    def get_inspect_for_image(
+        self, image: Union[str, ImageName], platform: Optional[str] = None
+    ) -> dict:
```

Comment:
How about reuse the type alias `ImageInspectionData`?

Reply:
Good idea, I'll update the corresponding RegistryClient method as well

Reply:
updated